runtime/compiler/compile/J9Compilation.cpp

```diff
@@ -614,8 +613,13 @@ J9::Compilation::canAllocateInline(TR::Node* node, TR_OpaqueClassBlock* &classIn
       TR_ASSERT(node->getSecondChild()->getOpCode().isLoadConst(), "Expecting const child \n");
 
       int32_t arrayClassIndex = node->getSecondChild()->getInt();
-
+#if defined(JITSERVER_SUPPORT)
```

Comment:
We can use the JITServer implementation for non-jitserver as well. It only needs the addition of TR_J9VMBase::getClassFromNewArrayTypeNonNull(int32_t arrayType).
For a cleaner delivery ** to master ** this can be done in a separate PR prior to this one.

Reply:
`TR_J9VMBase::getClassFromNewArrayTypeNonNull(int32_t arrayType).` is already added in the mirrored master PR #7016. I'll update `J9Compilation.cpp` using JITServer implementation.